New heat capacity function and example

mrshirts opened the pull request:
Revision of the heat capacity calculation function.  See calculate_heat_capacity.py for how to call.  Note that a new .nc must be located where the script ran that run, and the number of replicas and min/max temperature must correspond to what the replica exchange simulation was run with (we need a better way of passing that data to analysis scripts. I didn't not include an example .nc file to avoid cluttering the repository.

cwalker7 commented:
This seems to be working as expected - I tried a different example (24mer with 48 replica from 50K to 600K) with various intermediate temps, and got Cv plots similar to the one you showed yesterday. 

We should get rid of or update heat_capacity.py in that same example folder to avoid confusion.

mrshirts commented:
OK, I cleaned out that folder.

* I realized there was no way to control the plot, so I added a plot_file argument to the main function call to get that information down to the plot function
* .pdf with vector graphics is better than .png, so I switched to plotting .pdfs.
* I added text printout of the energies so they can be seen.
* I removed heat_capacity.py, helix.pdb and several plots for replica exchange that are no longer generated here.

mrshirts commented:
These changes seem to work OK, so I was about to merge, but now this branch seems to be including the expectation value PR (#18) as well. So I'll wait until you review that one before I merge this one into master.
